### ISSUE TITLE
Introduce Unmodifiable wrapper suitable for TreeMultimap

### ISSUE THREAD

**Issue by avakel** [google/guava#3143]
The [unmodifiableSortedSetMultimap method in Multimaps](https://github.com/google/guava/blob/07b4f00bc2cb5fcab1b96df8d186e499703eb50d/guava/src/com/google/common/collect/Multimaps.java#L930-L950) is currently the best way to get an unmodifiable view of a `TreeMultimap` but lacks two key features of `TreeMultimap`s:
1. Guaranteeing the sorting of the keys
1. Returning `NavigableSet`s and `NavigableMap`s from appropriate methods (in non-GWT)

**Comment by cpovirk** [comment]
We could probably take a patch for this if you're interested. The approach would probably be for the unmodifiable multimaps to call a method similar to the `unmodifiableCollectionSubclass` methods in classes like `Multimaps.CustomMultimap`.

**Comment by cpovirk** [comment]
Yes, sorry, I'm thinking just of guaranteeing that you can cast.
